src/acme/ui/ReportsPanel.java

```diff
@@ -178,32 +175,45 @@ private void printReport() {
     public void updateNameSel() {
         ArrayList<String> items = new ArrayList<>();
         // clear out the comboBox
-        nameSel.removeAllItems();
+        try {

```

Comment:
It does the check inside of updateNameSel now. This way we don't have to check it each time we call updateNameSel().